integrate opencast studio

darolfes opened the pull request:
This Request integrates [Opencast Studio](https://github.com/elan-ev/opencast-studio), a Web-based recording software, into Opencast.
Opencast Studio can be used by navigating to `<YOUR-OPENCAST>/studio` or using the new header in the opencast admin-ui.
Currently admin rights are needed to visit the page.

darolfes committed:
integrate opencast-studio

LukasKalbertodt commented:
Some context: we discussed how to best include OC Studio into this repository, but couldn't find a really satisfying result.

For one, OC Studio actually needs to get "compiled" when compiling OC (*). It's not possible to just include a compiled version because we need to configure `PUBLIC_URL` to be `/studio` (or whatever path we want). This environment variable has to be set during `npm build` and is empty by default (to be deployed at studio.opencast.org).

So how to include the source code of studio?
- From a purely technical standpoint I would prefer using a git submodule. But this would be the first submodule in this repo, thus probably confusing and annoying people not used to working with submodules.
- A similar solution is to have the build script `git clone` the studio repo. 
- Finally, what is implemented currently: we have a fake NPM package that has studio as dependency. That was NPM does all the downloading for us. 

None of these solutions seems particularly great to me. 

---

(*) While writing this comment, I thought about this statement again. It should be possible without too much hassle to automatically create a compiled version with `PUBLIC_URL=/studio`. This could be archived and uploaded (e.g. as a GitHub release). Then this module in this repository would just link to the archive and download it when building the module. 

We will talk about this and a couple other things here next week in Osnabrück. It's just a draft PR after all :)

gregorydlogan commented:
IMO a submodule here would be appropriate, and I don't think we should be afraid of making (more) use of Git's features just because some devs are not as familiar with it.  I think a clone + build would be cleaner than a fake npm package - the current setup seems like a hack to me, even if it works :)

lkiesow commented:
We decided to just go with downloading and including releases. It doesn't really make sense to rebuild the same version over and over again anyway ;-)

lkiesow commented on assemblies/karaf-features/src/main/feature/feature.xml:
```diff
@@ -242,6 +242,7 @@
     <bundle start-level="82">mvn:org.opencastproject/opencast-statistics-provider-random/${project.version}</bundle>
     <bundle start-level="82">mvn:org.opencastproject/opencast-statistics-service-api/${project.version}</bundle>
     <bundle start-level="82">mvn:org.opencastproject/opencast-statistics-service-impl/${project.version}</bundle>
+    <bundle start-level="82">mvn:org.opencastproject/opencast-studio/${project.version}</bundle>
```
We should include this in both the allinone and the admin distribution

lkiesow commented on modules/studio/pom.xml:
```diff
@@ -0,0 +1,103 @@
+          <execution>
+            <phase>initialize</phase>
+            <goals>
+              <goal>read-project-properties</goal>
```
Just wondering why we don't just specify these properties in lines 13+ of this `pom.xml` file? Wouldn't that be easier?

lkiesow commented on modules/studio/pom.xml:
```diff
@@ -0,0 +1,103 @@
+      <plugin>
+        <groupId>com.googlecode.maven-download-plugin</groupId>
+        <artifactId>download-maven-plugin</artifactId>
+        <version>1.5.0</version>
```
Can you push the version specification to the main `pom.xml` file?

lkiesow committed:
Minor Studio Patches

This patch

- Rebases the patch to `r/8.x`
- Adds security rules for `ROLE_STUDIO`
- Adds `ROLE_STUDIO` to the sadmin interface role selection so that
  this role can be assigned to users.
- Updates Stusio to the latest release

lkiesow commented:
I've reviewed this and added a patch that:

- Rebases the patch to `r/8.x`
- Adds security rules for `ROLE_STUDIO`
- Adds `ROLE_STUDIO` to the sadmin interface role selection so that
  this role can be assigned to users.
- Updates Stusio to the latest release

What's still missing is:

- The default workflow used does not match the one specified in the studio settings. None is sent right now and the global default is used. That's probably fine but then there should also be none specified in the settings. Otherwise, the default from the settings should be used.
- No ACL is set which causes problems if a non-admin user uses Studio. Studio should send an ACL specifying the user's role as having read and write access.

@LukasKalbertodt, can you fix that (at least the latter point) and update the version tomorrow morning? And then assign and ping @gregorydlogan once that's done?

lkiesow committed:
Update to Studio 2019-03-13

gregorydlogan commented on etc/security/mh_default_org.xml:
```diff
@@ -261,6 +261,9 @@
     <sec:intercept-url pattern="/export/**" method="GET" access="ROLE_ANONYMOUS" />
 
+    <!-- Enable admin access to Opencast Studio -->
+    <sec:intercept-url pattern="/studio/**" access="ROLE_STUDIO" />
```
No one, even the global admin, has this permission by default.

lkiesow replied:
The global admin can always access this. Did you get refusal as admin?

gregorydlogan replied:
Logged in as admin:opencast and got a 403.

lkiesow replied:
Interesting. Anyway, added `ROLE_ADMIN`.

gregorydlogan commented:
Just finished recording a test recording, waiting on processing.  In the meantime:

1) My screen was just black.  I could see my mouse, but that was it.  The preview was black, but the options to share tabs or windows appear to work correctly.  My resoltuion is 2560x1440, running in Chrome.
2) The "Recording" tab in the top right.  What's it for?  It looks to my naive eyes like that is indicating that it's actually recording - and the "http://localhsot:8080 is sharing your screen." warning at the bottom is partially occluding the red button.  I would expect that tab to either be invisible when not recording, or light up somehow when you *are* recording.

Will report back when the processing is done.

LukasKalbertodt commented:
> My screen was just black. I could see my mouse, but that was it. The preview was black, but the options to share tabs or windows appear to work correctly. My resoltuion is 2560x1440, running in Chrome.

What operating system (and in case of Linux: desktop environment) are you using? And I am understanding you correctly in that only the "share my complete screen" showed these problems? We at one point encountered an issue where capturing browser windows would result in a black image. It turned out that this was caused by privacy/DRM stuff in those browsers. One related term is "WebContent occlusion". You can read more about this in [this issue](https://github.com/elan-ev/opencast-studio/issues/197).

So I *assume* that this has nothing to do with Studio or with this integration PR. It's rather likely that this is an issue with your browser+desktop environment. 

> The "Recording" tab in the top right. What's it for? It looks to my naive eyes like that is indicating that it's actually recording - 

The "Recording" navigation link has a similar purpose to the other two navigation links. For example, if you visit the settings, you have to "get back" somehow. But I agree that the UX is not optimal (since most users spend the vast majority of time on that "Recording" page). A few months back, settings and the "about" page were modal dialogues -- an even worse solution IMO.  

> and the "http://localhsot:8080 is sharing your screen." warning at the bottom is partially occluding the red button. 

We can't do anything about that, as the warning is browser specific and we have no control over it. 

> I would expect that tab to either be invisible when not recording, or light up somehow when you are recording.

Mh. I understand the confusion. I guess in German it's less problematic as the noun "recording"/"Aufnahme" is not also the "act of recording" in German. So it's clear that this is not displaying a status. We will certainly think about how to improve the UX around that :)
But I guess this is not a blocker for this PR?

gregorydlogan commented:
Ok, so the core bits here work as expected, however I ran into a few exceptions.  Relatively certain they're either a result of, or missed parts of #1457 and #1458:

```
2020-03-13T13:19:27,840 | INFO  | (ConfigurablePublishWorkflowOperationHandler:337) - Published 7 elements of media package ff9ae9ef-3339-44df-b20e-b48ad0ed8df4 to publication channel internal
2020-03-13T13:19:33,027 | INFO  | (WaveformWorkflowOperationHandler:121) - Start waveform workflow operation for mediapackage 3fec5e84-93b1-42e2-ac1b-e49a4933338d
2020-03-13T13:19:33,028 | INFO  | (WaveformWorkflowOperationHandler:183) - Creating waveform extraction job for track 'a84562cf-77d3-4f83-a4fa-d8171040ff9a' in mediapackage '3fec5e84-93b1-42e2-ac1b-e49a4933338d'
2020-03-13T13:19:33,049 | INFO  | (WaveformWorkflowOperationHandler:183) - Creating waveform extraction job for track '599b5b34-75a9-4c16-9a21-52dd3aea6f3e' in mediapackage '3fec5e84-93b1-42e2-ac1b-e49a4933338d'
2020-03-13T13:19:33,096 | INFO  | (AssetManagerSnapshotWorkflowOperationHandler:91) - Take snapshot of media package ff9ae9ef-3339-44df-b20e-b48ad0ed8df4
2020-03-13T13:19:33,106 | INFO  | (AbstractAssetManager:242) - Creating new version 2 of media package ff9ae9ef-3339-44df-b20e-b48ad0ed8df4
2020-03-13T13:19:33,177 | INFO  | (AssetManagerWithMessaging:211) - Send update message for snapshot ff9ae9ef-3339-44df-b20e-b48ad0ed8df4, 2 to ActiveMQ
2020-03-13T13:19:38,087 | INFO  | (ServiceRegistryJpaImpl:2590) - State set to WARNING for current service org.opencastproject.waveform on host http://localhost:8080
2020-03-13T13:19:38,203 | ERROR | (AbstractJobProducer$JobRunner:344) - Error handling operation 'Waveform': java.lang.NullPointerException
	at org.opencastproject.waveform.ffmpeg.WaveformServiceImpl.getWaveformImageWidth(WaveformServiceImpl.java:441)
	at org.opencastproject.waveform.ffmpeg.WaveformServiceImpl.extractWaveform(WaveformServiceImpl.java:308)
	at org.opencastproject.waveform.ffmpeg.WaveformServiceImpl.process(WaveformServiceImpl.java:263)
	at org.opencastproject.job.api.AbstractJobProducer$JobRunner.call(AbstractJobProducer.java:314)
	at org.opencastproject.job.api.AbstractJobProducer$JobRunner.call(AbstractJobProducer.java:273)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

2020-03-13T13:19:38,215 | INFO  | (ServiceRegistryJpaImpl:2598) - State set to ERROR for current service org.opencastproject.waveform on host http://localhost:8080
2020-03-13T13:19:38,281 | ERROR | (AbstractJobProducer$JobRunner:344) - Error handling operation 'Waveform': java.lang.NullPointerException
	at org.opencastproject.waveform.ffmpeg.WaveformServiceImpl.getWaveformImageWidth(WaveformServiceImpl.java:441)
	at org.opencastproject.waveform.ffmpeg.WaveformServiceImpl.extractWaveform(WaveformServiceImpl.java:308)
	at org.opencastproject.waveform.ffmpeg.WaveformServiceImpl.process(WaveformServiceImpl.java:263)
	at org.opencastproject.job.api.AbstractJobProducer$JobRunner.call(AbstractJobProducer.java:314)
	at org.opencastproject.job.api.AbstractJobProducer$JobRunner.call(AbstractJobProducer.java:273)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

2020-03-13T13:19:43,183 | ERROR | (WorkflowOperationWorker:143) - Workflow operation 'operation:'waveform', position:18, state:'FAILED'' failed
org.opencastproject.workflow.api.WorkflowOperationException: Waveform extraction jobs for media package '3fec5e84-93b1-42e2-ac1b-e49a4933338d' have not completed successfully
	at org.opencastproject.workflow.handler.waveform.WaveformWorkflowOperationHandler.start(WaveformWorkflowOperationHandler.java:195) ~[?:?]
	at org.opencastproject.workflow.impl.WorkflowOperationWorker.start(WorkflowOperationWorker.java:193) ~[210:opencast-workflow-service-impl:8.0.0.SNAPSHOT]
	at org.opencastproject.workflow.impl.WorkflowOperationWorker.execute(WorkflowOperationWorker.java:116) [210:opencast-workflow-service-impl:8.0.0.SNAPSHOT]
	at org.opencastproject.workflow.impl.WorkflowServiceImpl.runWorkflowOperation(WorkflowServiceImpl.java:784) [210:opencast-workflow-service-impl:8.0.0.SNAPSHOT]
	at org.opencastproject.workflow.impl.WorkflowServiceImpl.process(WorkflowServiceImpl.java:1806) [210:opencast-workflow-service-impl:8.0.0.SNAPSHOT]
	at org.opencastproject.workflow.impl.WorkflowServiceImpl$JobRunner.call(WorkflowServiceImpl.java:2234) [210:opencast-workflow-service-impl:8.0.0.SNAPSHOT]
	at org.opencastproject.workflow.impl.WorkflowServiceImpl$JobRunner.call(WorkflowServiceImpl.java:2200) [210:opencast-workflow-service-impl:8.0.0.SNAPSHOT]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
	at java.lang.Thread.run(Thread.java:748) [?:?]
```

and

```
2020-03-13T13:23:27,996 | WARN  | (TimelinePreviewsServiceImpl:314) - Error creating timeline preview images for http://localhost:8080/files/mediapackage/3fec5e84-93b1-42e2-ac1b-e49a4933338d/9b400112-3934-492b-9369-657ddd1bf5c7/presenter___test_2.webm
org.opencastproject.mediapackage.MediaPackageException: Track http://localhost:8080/files/mediapackage/3fec5e84-93b1-42e2-ac1b-e49a4933338d/9b400112-3934-492b-9369-657ddd1bf5c7/presenter___test_2.webm does not have a duration
	at org.opencastproject.timelinepreviews.ffmpeg.TimelinePreviewsServiceImpl.generatePreviewImages(TimelinePreviewsServiceImpl.java:281) [182:opencast-timelinepreviews-ffmpeg:8.0.0.SNAPSHOT]
	at org.opencastproject.timelinepreviews.ffmpeg.TimelinePreviewsServiceImpl.process(TimelinePreviewsServiceImpl.java:340) [182:opencast-timelinepreviews-ffmpeg:8.0.0.SNAPSHOT]
	at org.opencastproject.job.api.AbstractJobProducer$JobRunner.call(AbstractJobProducer.java:314) [60:opencast-common:8.0.0.SNAPSHOT]
	at org.opencastproject.job.api.AbstractJobProducer$JobRunner.call(AbstractJobProducer.java:273) [60:opencast-common:8.0.0.SNAPSHOT]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
	at java.lang.Thread.run(Thread.java:748) [?:?]
2020-03-13T13:23:28,030 | INFO  | (ServiceRegistryJpaImpl:2590) - State set to WARNING for current service org.opencastproject.timelinepreviews on host http://localhost:8080
2020-03-13T13:23:28,061 | ERROR | (AbstractJobProducer$JobRunner:344) - Error handling operation 'TimelinePreview': org.opencastproject.serviceregistry.api.ServiceRegistryException: Error handling operation 'TimelinePreview'
	at org.opencastproject.timelinepreviews.ffmpeg.TimelinePreviewsServiceImpl.process(TimelinePreviewsServiceImpl.java:350)
	at org.opencastproject.job.api.AbstractJobProducer$JobRunner.call(AbstractJobProducer.java:314)
	at org.opencastproject.job.api.AbstractJobProducer$JobRunner.call(AbstractJobProducer.java:273)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.opencastproject.timelinepreviews.api.TimelinePreviewsException: org.opencastproject.mediapackage.MediaPackageException: Track http://localhost:8080/files/mediapackage/3fec5e84-93b1-42e2-ac1b-e49a4933338d/9b400112-3934-492b-9369-657ddd1bf5c7/presenter___test_2.webm does not have a duration
	at org.opencastproject.timelinepreviews.ffmpeg.TimelinePreviewsServiceImpl.generatePreviewImages(TimelinePreviewsServiceImpl.java:318)
	at org.opencastproject.timelinepreviews.ffmpeg.TimelinePreviewsServiceImpl.process(TimelinePreviewsServiceImpl.java:340)
	... 6 more
Caused by: org.opencastproject.mediapackage.MediaPackageException: Track http://localhost:8080/files/mediapackage/3fec5e84-93b1-42e2-ac1b-e49a4933338d/9b400112-3934-492b-9369-657ddd1bf5c7/presenter___test_2.webm does not have a duration
	at org.opencastproject.timelinepreviews.ffmpeg.TimelinePreviewsServiceImpl.generatePreviewImages(TimelinePreviewsServiceImpl.java:281)
	... 7 more

2020-03-13T13:23:28,067 | WARN  | (TimelinePreviewsServiceImpl:314) - Error creating timeline preview images for http://localhost:8080/files/mediapackage/3fec5e84-93b1-42e2-ac1b-e49a4933338d/9ccd615b-16c8-4075-ae3f-4775638e33a6/presentation___test_2.webm
org.opencastproject.mediapackage.MediaPackageException: Track http://localhost:8080/files/mediapackage/3fec5e84-93b1-42e2-ac1b-e49a4933338d/9ccd615b-16c8-4075-ae3f-4775638e33a6/presentation___test_2.webm does not have a duration
	at org.opencastproject.timelinepreviews.ffmpeg.TimelinePreviewsServiceImpl.generatePreviewImages(TimelinePreviewsServiceImpl.java:281) [182:opencast-timelinepreviews-ffmpeg:8.0.0.SNAPSHOT]
	at org.opencastproject.timelinepreviews.ffmpeg.TimelinePreviewsServiceImpl.process(TimelinePreviewsServiceImpl.java:340) [182:opencast-timelinepreviews-ffmpeg:8.0.0.SNAPSHOT]
	at org.opencastproject.job.api.AbstractJobProducer$JobRunner.call(AbstractJobProducer.java:314) [60:opencast-common:8.0.0.SNAPSHOT]
	at org.opencastproject.job.api.AbstractJobProducer$JobRunner.call(AbstractJobProducer.java:273) [60:opencast-common:8.0.0.SNAPSHOT]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
	at java.lang.Thread.run(Thread.java:748) [?:?]
2020-03-13T13:23:28,101 | INFO  | (ServiceRegistryJpaImpl:2598) - State set to ERROR for current service org.opencastproject.timelinepreviews on host http://localhost:8080
2020-03-13T13:23:28,138 | ERROR | (AbstractJobProducer$JobRunner:344) - Error handling operation 'TimelinePreview': org.opencastproject.serviceregistry.api.ServiceRegistryException: Error handling operation 'TimelinePreview'
	at org.opencastproject.timelinepreviews.ffmpeg.TimelinePreviewsServiceImpl.process(TimelinePreviewsServiceImpl.java:350)
	at org.opencastproject.job.api.AbstractJobProducer$JobRunner.call(AbstractJobProducer.java:314)
	at org.opencastproject.job.api.AbstractJobProducer$JobRunner.call(AbstractJobProducer.java:273)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.opencastproject.timelinepreviews.api.TimelinePreviewsException: org.opencastproject.mediapackage.MediaPackageException: Track http://localhost:8080/files/mediapackage/3fec5e84-93b1-42e2-ac1b-e49a4933338d/9ccd615b-16c8-4075-ae3f-4775638e33a6/presentation___test_2.webm does not have a duration
	at org.opencastproject.timelinepreviews.ffmpeg.TimelinePreviewsServiceImpl.generatePreviewImages(TimelinePreviewsServiceImpl.java:318)
	at org.opencastproject.timelinepreviews.ffmpeg.TimelinePreviewsServiceImpl.process(TimelinePreviewsServiceImpl.java:340)
	... 6 more
Caused by: org.opencastproject.mediapackage.MediaPackageException: Track http://localhost:8080/files/mediapackage/3fec5e84-93b1-42e2-ac1b-e49a4933338d/9ccd615b-16c8-4075-ae3f-4775638e33a6/presentation___test_2.webm does not have a duration
	at org.opencastproject.timelinepreviews.ffmpeg.TimelinePreviewsServiceImpl.generatePreviewImages(TimelinePreviewsServiceImpl.java:281)
	... 7 more

2020-03-13T13:23:28,176 | ERROR | (WorkflowOperationWorker:143) - Workflow operation 'operation:'timelinepreviews', position:45, state:'FAILED'' failed
org.opencastproject.workflow.api.WorkflowOperationException: Timeline previews jobs for media package '3fec5e84-93b1-42e2-ac1b-e49a4933338d' have not completed successfully
	at org.opencastproject.workflow.handler.timelinepreviews.TimelinePreviewsWorkflowOperationHandler.start(TimelinePreviewsWorkflowOperationHandler.java:186) ~[?:?]
	at org.opencastproject.workflow.impl.WorkflowOperationWorker.start(WorkflowOperationWorker.java:193) ~[210:opencast-workflow-service-impl:8.0.0.SNAPSHOT]
	at org.opencastproject.workflow.impl.WorkflowOperationWorker.execute(WorkflowOperationWorker.java:116) [210:opencast-workflow-service-impl:8.0.0.SNAPSHOT]
	at org.opencastproject.workflow.impl.WorkflowServiceImpl.runWorkflowOperation(WorkflowServiceImpl.java:784) [210:opencast-workflow-service-impl:8.0.0.SNAPSHOT]
	at org.opencastproject.workflow.impl.WorkflowServiceImpl.process(WorkflowServiceImpl.java:1806) [210:opencast-workflow-service-impl:8.0.0.SNAPSHOT]
	at org.opencastproject.workflow.impl.WorkflowServiceImpl$JobRunner.call(WorkflowServiceImpl.java:2234) [210:opencast-workflow-service-impl:8.0.0.SNAPSHOT]
	at org.opencastproject.workflow.impl.WorkflowServiceImpl$JobRunner.call(WorkflowServiceImpl.java:2200) [210:opencast-workflow-service-impl:8.0.0.SNAPSHOT]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
	at java.lang.Thread.run(Thread.java:748) [?:?]
2020-03-13T13:23:28,204 | INFO  | (TagWorkflowOperationHandler:92) - Retagging mediapackage elements
```

gregorydlogan commented:
> > My screen was just black. I could see my mouse, but that was it. The preview was black, but the options to share tabs or windows appear to work correctly. My resoltuion is 2560x1440, running in Chrome.
> 
> What operating system (and in case of Linux: desktop environment) are you using? And I am understanding you correctly in that only the "share my complete screen" showed these problems? We at one point encountered an issue where capturing browser windows would result in a black image. It turned out that this was caused by privacy/DRM stuff in those browsers. One related term is "WebContent occlusion". You can read more about this in [this issue](https://github.com/elan-ev/opencast-studio/issues/197).

Debian 10, and Gnome.  It'd imagine it was DRM/privacy related since everything else appears to be working.  I have multiple browser windows open, or it could simply be that Chrome doesn't like me :)

> Mh. I understand the confusion. I guess in German it's less problematic as the noun "recording"/"Aufnahme" is not also the "act of recording" in German. So it's clear that this is not displaying a status. We will certainly think about how to improve the UX around that :)
> But I guess this is not a blocker for this PR?

Nope, not a blocker, just giving feedback as someone unfamiliar with using the tool.

lkiesow committed:
Switch to fast workflow

mliradelc commented:
1. Why this was merged if the build was failing?
2. Where is the documentation? Any new feature MUST have their documentation.
